ball.js

```diff
@@ -0,0 +1,39 @@
+class Ball {
+  constructor(ctx, x, y, radius, color, speed) {
+    this.ctx = ctx;
+    this.x = x;
+    this.y = y;
+    this.radius = radius;
+    this.color = color;
+    this.speed = speed;
+    this.xDirection = 0;
+    this.yDirection = 1;
+  }
+
+  move() {
+    this.x += this.xDirection * this.speed;
+    this.y += this.yDirection * this.speed;
+    this.updateEdges();
+  }
+
+  updateEdges() {
```

Comment:
whats this for? top/bottom/left/right are never defined in constructor. if they're only used for collision, why not make these functions to return top/bottom/left/right for the ball?

Reply:
top/bottom/left/right is used for collision, where the ball edges meet canvas edges/ brick edges. can you clarify what you mean by making functions return these properties?

Reply:
these are just properties that don't really need to belong in the object. instead, you can create getter functions that return this for you

getLeftEdge(), getRightEdge(), getTopEdge(), getBottomEdge()